common/archiver/s3store/query_parser_test.go

```diff
@@ -188,6 +187,10 @@ func (s *queryParserSuite) TestParsePrecision() {
 	}
 }
 
+func ptr[t any](v t) *t {
```

Comment:
Is it used anywhere else?

Reply:
I swapped `t` for `T` but didn't put this into our utility library. I'd prefer that we  address things regularly (using `&stackVar`) instead of this in our normal code; I added this to make tests simpler to write